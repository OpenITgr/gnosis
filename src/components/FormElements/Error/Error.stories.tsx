import React from "react";
import { Story } from "@storybook/react";
import ErrorComponent from "./Error";
import { Input } from "@components";

export default {
  title: "components/Form Elements/Error",
  component: ErrorComponent,
  parameters: {
    controls: { hideNoControlsWarning: true },
  },
};

export const Error: Story = () => (
  <ErrorComponent>
    <p>Something went wrong!</p>
  </ErrorComponent>
);

export const InputError: Story = () => (
  <div style={{ maxWidth: 500 }}>
    <Input
      id="inline-error-example"
      label="Attention"
      status="error"
      placeholder="Pay attention to the error below"
    />
    <ErrorComponent.InputError>Some error</ErrorComponent.InputError>
  </div>
);
