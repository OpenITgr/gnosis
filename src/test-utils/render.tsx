/* eslint-disable import/export */
import React, { FC, ReactElement } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options?: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
