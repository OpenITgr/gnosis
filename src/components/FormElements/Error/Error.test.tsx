import React from "react";
import { Error, InputError } from "./Error.stories";
import { render, screen } from "@test-utils/render";

describe("Forms: <Error />", () => {
  it("render correctly", () => {
    render(<Error />);
    const errorContainer = screen.getByText("Something went wrong!");

    expect(errorContainer).toHaveTextContent("Something went wrong!");
  });

  it("matches snapshot", () => {
    const { container } = render(<Error />);

    expect(container).toMatchSnapshot();
  });
});

describe("Forms: <Error.InputError />", () => {
  it("renders correctly", () => {
    render(<InputError />);
    const errorSpan = screen.getByText("Some error");

    expect(errorSpan).toHaveTextContent("Some error");
  });

  it("matches snapshot", () => {
    const { container } = render(<InputError />);

    expect(container).toMatchSnapshot();
  });
});
