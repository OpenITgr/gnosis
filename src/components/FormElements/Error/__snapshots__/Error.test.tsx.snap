// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forms: <Error /> matches snapshot 1`] = `
<div>
  <article
    class="css-gbf80r-errorContainer-errorContainer"
  >
    <p>
      Something went wrong!
    </p>
  </article>
</div>
`;

exports[`Forms: <Error.InputError /> matches snapshot 1`] = `
<div>
  <div
    style="max-width: 500px;"
  >
    <div
      class="error css-38lqwp-inputContainerBaseStyles-inputContainerBaseStyles-inputBaseStyles-inputBaseStyles-inputContainer-inputContainer"
    >
      <label
        class="css-xs6o3g-label-label"
        for="inline-error-example"
      >
        Attention
      </label>
      <div
        class="input-wrapper"
      >
        <input
          id="inline-error-example"
          placeholder="Pay attention to the error below"
        />
      </div>
    </div>
    <span
      class="css-1b2rqxw-inputError-inputError"
    >
      Some error
    </span>
  </div>
</div>
`;
